metecho/api/models.py

```diff
@@ -1066,6 +1079,33 @@ def finalize_refresh_org(self, *, error=None, originating_user_id):
             )
             self.queue_delete(originating_user_id=originating_user_id)
 
+    def queue_reassign(self, *, new_user, originating_user_id):
+        from .jobs import user_reassign_job
+
+        self.currently_refreshing_org = True
+        self.deleted_at = None
+        self.save()
+        self.notify_list_changed(
+            type_="SCRATCH_ORG_RECREATE", originating_user_id=originating_user_id,
+        )
+        user_reassign_job.delay(
+            self, new_user=new_user, originating_user_id=originating_user_id
+        )
+
+    def finalize_reassign(self, *, error=None, originating_user_id):
+        self.currently_reassigning_user = False
+        self.save()
+        if error is None:
+            self.notify_changed(
+                type_="SCRATCH_ORG_REASSIGN", originating_user_id=originating_user_id
+            )
+        else:
+            self.notify_scratch_org_error(
+                error=error,
+                type_="SCRATCH_ORG_REASSIGN_FAILED",
+                originating_user_id=originating_user_id,
+            )
```

Comment:
@wlonk I'm trying to think through what we should do on the error case here... Probably soft-delete the org?